File: Level 1/Stack Stabilization 1/python/Stack_Stabilization_1.py
from typing import List

def getMinimumDeflatedDiscCount(N: int, R: List[int]) -> int:
    count = 0
    
    for i in range(N - 2, -1, -1):
        if R[i] >= R[i + 1]:
            R[i] = R[i + 1] - 1
            if R[i] == 0:
                return -1
            count += 1
    
    return count

File: Level 1/Stack Stabilization 1/javascript/StackStabilization1.js
/**
 * @param {number} N
 * @param {number[]} R
 * @return {number}
 */
function getMinimumDeflatedDiscCount(N, R) {
    count = 0
    for (i = N - 2; i >= 0; i--) {
        if (R[i] >= R[i+1]) {
            R[i] = R[i+1] - 1
            if (R[i] == 0) {
                return -1
            }
            count++
        }
    }
    
    return count
}